app/Silex/UiTPAS/UiTPASIncomingEventServicesProvider.php

```diff
@@ -5,30 +5,39 @@
 namespace CultuurNet\UDB3\Silex\UiTPAS;
 
 use Broadway\EventHandling\EventBus;
+use CultuurNet\UDB3\Container\AbstractServiceProvider;
 use CultuurNet\UDB3\Deserializer\SimpleDeserializerLocator;
 use CultuurNet\UDB3\Broadway\AMQP\EventBusForwardingConsumerFactory;
 use CultuurNet\UDB3\ApiName;
-use CultuurNet\UDB3\Silex\Container\HybridContainerApplication;
 use CultuurNet\UDB3\Error\LoggerFactory;
 use CultuurNet\UDB3\Error\LoggerName;
 use CultuurNet\UDB3\UiTPAS\Event\Event\EventCardSystemsUpdatedDeserializer;
 use CultuurNet\UDB3\UiTPAS\Event\Event\PricesUpdatedDeserializer;
 use CultuurNet\UDB3\UiTPAS\Event\EventProcessManager;
 use CultuurNet\UDB3\UiTPAS\Label\InMemoryUiTPASLabelsRepository;
 use Ramsey\Uuid\UuidFactory;
-use Silex\Application;
-use Silex\ServiceProviderInterface;
 use CultuurNet\UDB3\StringLiteral;
 
-class UiTPASIncomingEventServicesProvider implements ServiceProviderInterface
+final class UiTPASIncomingEventServicesProvider extends AbstractServiceProvider
 {
-    public function register(Application $app): void
+    protected function getProvidedServiceNames(): array
     {
-        $app['amqp.uitpas_event_bus_forwarding_consumer'] = $app->share(
-            function (HybridContainerApplication $app) {
+        return [
+            'amqp.uitpas_event_bus_forwarding_consumer',
+            'uitpas_event_process_manager',
+        ];
+    }
+
+    public function register(): void
+    {
+        $container = $this->getContainer();
+
+        $container->addShared(
+            'amqp.uitpas_event_bus_forwarding_consumer',
+            function () use ($container) {
                 // If this service gets instantiated, it's because we're running the AMQP listener for UiTPAS messages
                 // so we should set the API name to UiTPAS listener.
-                $app['api_name'] = ApiName::UITPAS_LISTENER;
+                $container->addShared('api_name', ApiName::UITPAS_LISTENER);
```

Comment:
Q: I wonder if this will overwrite a previously set api_name. With the previous implementation, it will overwrite it because of the array access. A quick glance at the current implementation of the container makes me think this will just add a new definition to the container, without erasing the previous one.

Reply:
<img width="470" alt="Screenshot 2022-10-10 at 18 55 45" src="https://user-images.githubusercontent.com/712929/194917559-b0b09519-6778-4856-b3c4-0f295ec94885.png">
<img width="525" alt="Screenshot 2022-10-10 at 18 55 31" src="https://user-images.githubusercontent.com/712929/194917562-32b59728-2363-4b10-a156-9a5f33561a6b.png">

Tested it out, it keeps the original value, so this change will likely break something.

Reply:
Some possible solutions I could think of off the top of my head (better ones may also be possible):

- Using global variables as a temporary fix
- Using environment variables (using `putenv()` so we can set them in the code)
- Adding these kinds of variables to the `config` array? (Not sure if possible, maybe by reference?)

Reply:
\+ Even if the `addShared()` could overwrite another service in the League container, it would not work in this case because the `api_name` is set on `$app` here, which means it will exist in two service containers (in this PR) with a different value in each: https://github.com/cultuurnet/udb3-silex/blob/7eeb838d1a925c1bd72a585764edcd8091fc4af3/bootstrap.php#L98